### ISSUE TITLE
Input Improvements

### ISSUE THREAD

**Issue by lasers** [libretro/RetroArch#3796]
Add / Fix the toggle option for... `Modifiers ── slowmotion ── shader ── easter egg`

- [ ] Improvement
* Ignore all key events when using a modifier key.

  When using keyboard shortcuts unrelated to the game, RetroArch will act like it ought to read them. This results in lot of accidental or unwanted behaviors.
  1. I switched or moved window/workspace not knowing I unpaused the game right away in the background. (key:p)
  2. I applied next or prev shaders many times enough to investigate a minor UI bug with shader keys. I already made an issue report. (key:n and key:m)
  3. I toggled fastforward when I used spacebar with modifier on something trivial... Not expecting to get back into a fast game when I unpaused. (key:space)
  4. I didn't enjoy resetting the game. I unbinded that one right away. (key:h)
  5. Some other insignificant keys I toggled. (key:k, key:l, etc)

- [ ] Improvement

  _I BELIEVE I FOUND A SECRET IN RETROARCH. CALL IT AN EASTER EGG OR A CHEAT CODE IF YOU WILL! THIS GEM IS BURIED DEEP IN THE CODE. READY? IT'S A HIDDEN BIND FOR SEARCH._

  As I can see so far, it is a hush-hush secret option not documented anywhere in default or fresh retroarch.cfg. Toggling `key:slash` or `key:s` will activate search prompt in RetroArch with no option to change either key in the config. I would like to stay with `key:slash` for search and leave `key:s` unbind for something else one day.

**Comment by fr500** [comment]
Yes, if you set input_enable_hotkey to something it ignores all the "core" inputs.

**Comment by lasers** [comment]
- [ ] Are we missing `User __ Bind Default All` for `Game Menu --> Controls`? After messing a simple remap, it is not possible to to perform "controller reset --hard" without remembering what the original buttons were. 

- [ ] Remove "Auto" -- It's annoying after a while and takes up so much space. Pick a good symbol, asterisk, or any other fitting character.

**Comment by fr500** [comment]
No point in adding bind all that. Just use start to reset the bindings.
I'll think about the auto thing

**Comment by lasers** [comment]
Ah right. You should remove `User 1 Bind Default All` from `Inputs` too because that technically would work fine too, but yes, I forget about that. Thanks. 😄

**Comment by fr500** [comment]
It's not the same. Input menu takes INPUT. The other one is multiple selection.
I'm not to keen on the sarcasm.

**Comment by lasers** [comment]
No sarcasm. If that's technically true, then I'm pointing it out. Simple as that. I'm also trying to understand the differences between multiple selections as I didn't want to ask something that minor.

For instance, it seems like '(btn)' only comes in play when it's used with 'Auto', but in User 2, it's not there. Does it mean that `(btn)` is necessary / excessive... especially when the left column already say button? What's the difference between (Key: left) and (left)? 

I also can't help that Github's :smile: looks sarcasm as hell. We need to take it up with them. Just read my statement again without the smile and you might see it could be just genuine. I'll just use :-) next time.

**Comment by lasers** [comment]
Subject: Re: Sarcastic emoji                                                                                           
                                                                                                                       
Hi there,                                                                                                                                                                                                                                     
Thanks for writing in! I can't make any promises, but I'll let the team know you'd like to see us adopt a more neutral 
+set of Emoji on the site.                                                                                             
                                                                                                                       
Please let me know if there's anything else I can do for you in the meantime.                                          
                                                                                                                       
All the best,                                                                                                          
Nick                                                                                                                   
GitHub Support                                                                                                         
                                                                                                                       
> Hi there. I can't use emoji nicely without people thinking it was a sarcastic comment because of the emojis.         
>                                                                                                                      
> While this issue may seems minor, I'm writing comments nicely as possible then I end it with a friendly smile only to send out something in sarcastic tone because of that emoji. The people will then react to what I said although that was not my intentional. I'm probably not the only one doing that.                                                     
>                                                                                                                      
> Here's what I wrote in my issue: https://github.com/libretro/RetroArch/issues/3796                                   
>                                                                                                                      
> I wanted to see if we can opt for Tango Smileys instead... as it's more neutral.                                     
> https://github.com/justnorris/tango-smileys-extended                                                                 
>                                                                                                                      
> Thank you for reading. :-)

**Comment by lasers** [comment]
- [ ] Toggling "Menu Swap OK & Cancel Buttons" behaves *slightly* like a jerk. Fix? Ignore?

**Comment by fr500** [comment]
How do you expect us to understand what does that even mean?
Really dude. I already told you to explain better. We need the expected result, and the current result at the very least.

I don't like having to guess what you mean and reading your "gotta poop" issue descriptions.

**Comment by lasers** [comment]
Current result: Acts a little funny... Jerk movements... Shaky... Wobbly... Seizure... Glitchy... Sticky.... Stubborn.  Stuck in same place sometimes.

Expected result: Just switch over like all other buttons.

Solve: Toggle it some more before it switch over.

Gotta poop is a step. Follow the steps and see if you can reproduce that same behavior.

**Comment by fr500** [comment]
You could have said...
Menu Swap action keeps repeating itself until you release the key.

You said: acts slightly like a jerk.

There is a big difference.

**Comment by lasers** [comment]
Ah. The issue appears when you're using the keyboard. There are zero issue with the controller. I didn't view the problem as the button repeating itself... just jerky / glitchy movements. Anyway, yes.... Good catch. Have a good day.

**Comment by lasers** [comment]
- [ ] Translate "[%s]\npress keyboard or joypad\n(timeout %d %s)"

- [ ] Change default bind_timeout from 5 to 10 to give users more time. I think this time may be somewhat aggressive? Discussion? Users need to read lines first before doing anything... leaving us with maybe 3 seconds.

- [ ] Add 1 second delay after capturing a bind to stop sensitive controllers from capturing the next bind immediately.

- [ ] Change the order of binds to stop throw people off with next steps.
```
{ "up", "down", "left", "right",
   "start", "select", "b", "y",
   "x", "a", "l", "r", "l2", "r2",
   "l3", "r3", "l_x", "l_y", "r_x", "r_y" };
```

**Comment by rgpires** [comment]
Feature Request

Would it be possible to add an option in retroarch to bind analog sticks direction as buttons? (at the same time as having actual button mapped)
Like, you move the right analog up and it works like pressing Y button, down would be A....
I know theres the option to use analog as digital but it works only for directions.
I ask that because it could benefit those who have physical impairments and share the controller with those who dont.
